planner/core/collect_column_stats_usage_test.go

```diff
@@ -209,13 +333,12 @@ func TestCollectPredicateColumns(t *testing.T) {
 		require.NoError(t, err, comment)
 		lp, ok := p.(plannercore.LogicalPlan)
 		require.True(t, ok, comment)
-		// We check predicate columns twice, before and after logical optimization. Some logical plan patterns may occur before
-		// logical optimization while others may occur after logical optimization.
-		// logutil.BgLogger().Info("before logical opt", zap.String("lp", plannercore.ToString(lp)))
-		checkPredicateColumns(lp, tt.res, comment)
-		lp, err = plannercore.LogicalOptimize(ctx, builder.GetOptFlag(), lp)
+		flags := builder.GetOptFlag()
+		// JoinReOrder may need columns stats so collecting hist-needed columns must happen before JoinReOrder.
+		// Hence we disable JoinReOrder and PruneColumnsAgain here.
+		flags &= ^(uint64(1<<13) | uint64(1<<14))
+		lp, err = plannercore.LogicalOptimize(ctx, flags, lp)
```

Comment:
Can we use the constant `flagXXXX` instead of using `1<<13`?

Reply:
I move the tests from `core_test` to `core` and use `flagXXX`.